src/express/commands/shadow.js

```diff
@@ -0,0 +1,81 @@
+import Web3 from 'web3'
+import { loadDevnetConfig, splitToArray } from '../common/config-utils'
+import { maxRetries, runSshCommand } from '../common/remote-worker'
+import { stopServices } from './cleanup'
+
+async function initWeb3(provider) {
+  return new Web3(provider)
+}
+
+export async function shadow(targetBlock) {
+  require('dotenv').config({ path: `${process.cwd()}/.env` })
+  const doc = await loadDevnetConfig('remote')
+  const borUsers = splitToArray(doc.devnetBorUsers.toString())
+  const providerToNodeIp = new Map()
+  let ip
+  let shadowBorChainId
+  let latestBlock
+  let provider
+  const providers = []
+
+  for (let i = 0; i < doc.devnetBorHosts.length; i++) {
+    ip = `${borUsers[i]}@${doc.devnetBorHosts[i]}`
+    provider = await initWeb3(`ws://${doc.devnetBorHosts[i]}:8546`)
```

Comment:
is this port open?

Reply:
Yep, we see here: https://github.com/maticnetwork/matic-cli/blob/f9706c3a6cbbf674094d61a854dd659833c18eb7/variables.tf#L86